moveit_ros/moveit_servo/include/moveit_servo/utilities.h

```diff
@@ -105,4 +105,28 @@ void transformTwistToPlanningFrame(geometry_msgs::msg::TwistStamped& cmd, const
  */
 geometry_msgs::msg::Pose poseFromCartesianDelta(const Eigen::VectorXd& delta_x,
                                                 const Eigen::Isometry3d& base_to_tip_frame_transform);
+
+/**
+ * @brief Decrease robot position change and velocity, if needed, to satisfy joint velocity limits
+ * @param joint_model_group Active joint group. Used to retrieve limits
+ * @param publish_period Period of the servo loop
+ * @param joint_state The command that will go to the robot
+ * @param override_velocity_scaling_factor Option if the user wants a constant override of the velocity scaling
+ *        a value greater than 0 will override the internal calculations done by getVelocityScalingFactor
+ */
+void enforceVelocityLimits(const moveit::core::JointModelGroup* joint_model_group, const double publish_period,
+                           sensor_msgs::msg::JointState& joint_state,
+                           const double override_velocity_scaling_factor = 0.0);
+
+/** \brief Avoid overshooting joint limits
+ * @param joint_state The joint state to be checked
+ * @param joint_limit_margin The allowed margin for joint limit
```

Comment:
Suggesting this b/c the user may be confused whether the margin is over the limit or prior to the limit.

```suggestion
 * @param joint_limit_margin The allowed margin for joint limit. This is a buffer, prior to the actual limit.
```